src/lib/mongoBackend/mongoSubscribeContext.cpp

```diff
@@ -133,11 +133,30 @@ HttpStatusCode mongoSubscribeContext
 
     /* Build attributes array */
     BSONArrayBuilder attrs;
-    for (unsigned int ix = 0; ix < requestP->attributeList.size(); ++ix) {
-        attrs.append(requestP->attributeList[ix]);
+    for (unsigned int ix = 0; ix < requestP->attributeList.size(); ++ix)
+    {
+      attrs.append(requestP->attributeList[ix]);
     }
     sub.append(CSUB_ATTRS, attrs.arr());
 
+
+    //
+    // StringFilter in Scope?
+    //
+    Restriction    restriction;
+    StringFilter*  stringFilterP = NULL;
+
+    for (unsigned int ix = 0; ix < requestP->restriction.scopeVector.size(); ++ix)
```

Comment:
True, that should work too, and removes code ans is faster in run-time.
I will look into it but I really think you're right